addons/web/models/models.py

```diff
@@ -1046,7 +1046,11 @@ def onchange(self, values: dict, field_names: list[str], fields_spec: dict):
         done = set()
 
         # mark fields to do as modified to trigger recomputations
-        protected = [self._fields[fname] for fname in field_names]
+        protected = [
+            field
+            for fname in field_names
+            for field in self.pool.field_computed.get(self._fields[fname], ())
```

Comment:
You have to follow what's done in the ORM itself. `field_computed.get(field)` may be empty in case the field is not computed but has an inverse method.  The right expression should be `field_computed.get(field, [field])` or `field_computed.get(field) or [field]`.